### PR TITLE
Categories Integration with API

### PR DESCRIPTION
**Pull request by Royal-lobster**
[Draft] Closes https://github.com/EveripediaNetwork/issues/issues/128

### CODE REVIEW

**Comment by kesar** [comment]
take a look to: https://github.com/EveripediaNetwork/ep-ui/tree/main/src/services/wikis to save graphql queries into the store

**Comment by kesar** [comment]
i think if we use an approach similar to: https://github.com/EveripediaNetwork/ep-ui/blob/main/src/pages/wiki/%5Bslug%5D.tsx#L31

we can get categories from SSR which would improve speed (hopefully) and SEO

**Comment by Royal-lobster** [comment]
> i think if we use an approach similar to: https://github.com/EveripediaNetwork/ep-ui/blob/main/src/pages/wiki/%5Bslug%5D.tsx#L31
> 
> & https://github.com/EveripediaNetwork/ep-ui/blob/main/src/pages/wiki/%5Bslug%5D.tsx#L17
> 
> we can get categories from SSR which would improve speed (hopefully) and SEO

Ohh that's awesome ! Thanks for pointing !

**Comment by kesar** [comment]
btw this needs to be done also for the categories menu in navbar and for the selection in the create wiki page

we should also remove the dummy old data once its done 👍🏻

File: src/components/Layout/Navbar/DesktopNav.tsx [hunk 1]
```diff
@@ -3,10 +3,40 @@ import { HStack } from '@chakra-ui/react'
...
   const [visibleMenu, setVisibleMenu] = useState<number | null>(null)
+  const [navData, setNavData] = useState<NavItem[]>(NAV_ITEMS)
+  const { data: topCategoriesLinks } = useGetTopCategoriesLinksQuery()
+
+  // Update nav data when top categories links are loaded
+  useEffect(() => {
+    if (topCategoriesLinks) {
+      const topCategories = topCategoriesLinks.map(
+        ({ title, id, icon }, i) => ({
+          id: parseInt(`10${i}${2}`, 10),
```
Comment (kesar) [inline src/components/Layout/Navbar/DesktopNav.tsx:20]: why is this? :O

File: src/pages/categories/index.tsx [hunk 2]
```diff
@@ -1,46 +1,66 @@
...
+        <SimpleGrid
+          columns={{ base: 1, sm: 2, lg: 3 }}
+          width="min(90%, 1200px)"
+          mx="auto"
+          my={12}
+          gap={8}
+        >
+          {data?.map((category, i) => (
+            <CategoryCard
+              key={category.id || sampleCategories[i].id}
```
Comment (kesar) [inline src/pages/categories/index.tsx:44]: no need to keep compatibility from sample

File: src/components/Layout/Navbar/MobileNav.tsx [hunk 3]
```diff
@@ -39,6 +40,34 @@ const MobileNav = ({ toggleWalletDrawer, setHamburger }: MobileNavType) => {
...
+  // Update nav data when top categories links are loaded
+  useEffect(() => {
+    if (topCategoriesLinks) {
+      const topCategories = topCategoriesLinks.map(
+        ({ title, id, icon }, i) => ({
+          id: parseInt(`10${i}${2}`, 10),
+          label: title || id,
+          href: `/categories/${id}`,
+          hasImage: true,
+          icon,
```
Comment (kesar) [inline src/components/Layout/Navbar/MobileNav.tsx:55]: I am expecting that Icon from the API will be a string like "BsBriefcaseFill" and then we will use that to put the Icon in the navbar. Not sure if that's how its working in this PR

**Comment by kesar** [comment]
image


I added more dummy data into the db that way we can see the loading of images / icons / etc 👍🏻

File: src/components/Layout/Navbar/MobileNav.tsx (hunk 3 above)
Comment (Royal-lobster) [reply src/components/Layout/Navbar/MobileNav.tsx:55]: Yes i am wondering how the icons are handled so i left it blank 😅. The string would work well with dynamic import so i will implement it in next commit !

File: src/pages/categories/index.tsx (hunk 2 above)
Comment (Royal-lobster) [reply src/pages/categories/index.tsx:44]: ohh yes. before i implemented the ui with slugs so i changed the implementation to use IDs. just wanted to test it out so i modified the sample data temporary.

File: src/components/Layout/Navbar/DesktopNav.tsx (hunk 1 above)
Comment (Royal-lobster) [reply src/components/Layout/Navbar/DesktopNav.tsx:20]: I was playing with RTK query 😅 the query fetches the top 10 categories from the database and injections in nav items data on client side to show in UI. That should be done on serverside using SSR so i am searching for elegant way to implement it. that should be sorted in my next commit 😅

File: src/data/CategoriesData.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@ import {
 
 export const sampleCategories: CategoryDataType = [
```
[the remark this answers is not in the document]
Comment (Royal-lobster) [reply src/data/CategoriesData.ts:16]: Yes. we can remove it after fully integrating with API.

**Comment by Royal-lobster** [comment]
> image
> 
> I added more dummy data into the db that way we can see the loading of images / icons / etc 👍🏻

That's awesome ! makes it easier to play and integrate ! Thanks 😊

File: src/data/CategoriesData.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kesar) [reply src/data/CategoriesData.ts]: I think we can remove it before merging right? (in this PR)

[the remark this answers is not in the document]
Comment (Royal-lobster) [reply src/data/CategoriesData.ts]: Yes ! i also changed usage of sample data inside create-wiki in highlight modal to API and deleted the sample data. Now its fully integrated

**Review by kesar — APPROVED** [review]

**Pull request merged**